File: Sorting/Insertion Sort/insertionsort.c
// Program to sort the array using Insertion Sort

#include<stdio.h>
#define maxSize 100

// Function to sort the array
void sortInsertion(int array[], int size)
{
    int temp;
    for(int i=1; i<size; i++)
    {
        temp = array[i];
        int j = i - 1;

        while(j >= 0 && array[j] > temp)
        {
            array[j+1] = array[j];
            j--;
        }

        array[j+1] = temp;
    }

    // Printing the sorted array
    printf("The sorted array: \n");
    for(int i=0; i<size; i++)
    {
        printf("%d ",array[i]);
    }
}

void main()
{
    int array[maxSize], size;

    // Scanning the size of the array
    printf("Enter the number of elements of array: ");
    scanf("%d",&size);

    // Scanning the elements of array
    printf("Enter the %d elements of array: \n",size);
    for(int i=0; i<size; i++)
    {
        scanf("%d",&array[i]);
    }

    // Calling the sortInseriton function
    sortInsertion(array,size);
}

File: Sorting/Insertion Sort/insertion_sort.py
# Program to perform insertion sort

# Function to scan the list
def scan_list(list_1, size):
    print(f"Enter the {size} elements of the array:")
    for i in range(size):
        element = int(input(f"Element [{i}]: "))
        list_1.append(element)


# Function to print the list
def print_list(list_1):
    print(list_1)


# Function to perform insertion sort
def insertion_sort(list_1, size):
    for i in range(size):
        temp = list_1[i]
        j = i - 1
        while j >= 0 and list_1[j] > temp:
            list_1[j+1] = list_1[j]
            j -= 1
        list_1[j+1] = temp


# Driver Function
if __name__ == '__main__':
    list_1 = []
    size = int(input("Enter the number of elements of the list: "))
    scan_list(list_1, size)
    print_list(f"List: {list_1}")
    insertion_sort(list_1, size)
    print_list(f"Sorted List: {list_1}")
